### PR TITLE
[DOCS] one_hot docs missing

### PR DESCRIPTION
**Pull request by krishnakalyan3**
one_hot docs is missing [here](https://pytorch.org/docs/master/nn.html#one-hot).

I dug around and could not find a way to get this working properly. 

cc: @zou3519 @zasdfgbnm
Could you please let me know how to proceed further?

Ref
#16508
#15208

### CODE REVIEW

**Comment by krishnakalyan3** [comment]
I also guess the example need to be changed to   
  
```>>> torch.one_hot(torch.arange(0, 5) % 3) 
to something like
F..one_hot(torch.arange(0, 5) % 3)
```

File: torch/nn/functional.py [hunk 1]
```diff
@@ -2834,7 +2834,7 @@ def pairwise_distance(x1, x2, p=2., eps=1e-6, keepdim=False):
 """)
 
 
-one_hot = _add_docstr(torch._C._nn.one_hot, r"""
+one_hot = _add_docstr(torch._C._nn.functional.one_hot, r"""
```
Comment (vishwakftw) [inline torch/nn/functional.py]: This is an incorrect change. I think the right change would be to move `torch.nn.utils.one_hot` from Utilities section to the Functional section and under Sparse Functions in this [file](https://github.com/pytorch/pytorch/blob/master/docs/source/nn.rst). This should allow `one_hot` to be discoverable in the docs.

Comment (vishwakftw) [reply torch/nn/functional.py]: Something like this:
```diff
diff --git a/docs/source/nn.rst b/docs/source/nn.rst
index f552ebb41..09fb41956 100644
--- a/docs/source/nn.rst
+++ b/docs/source/nn.rst
@@ -794,11 +794,6 @@ Utilities
 
 .. autofunction:: torch.nn.utils.remove_spectral_norm
 
-:hidden:`one_hot`
-~~~~~~~~~~~~~~~~~
-
-.. autofunction:: torch.nn.utils.one_hot
-
 
 .. currentmodule:: torch.nn.utils.rnn
 
@@ -1171,6 +1166,11 @@ Sparse functions
 
 .. autofunction:: embedding_bag
 
+:hidden:`one_hot`
+~~~~~~~~~~~~~~~~~
+
+.. autofunction:: one_hot
+
 Distance functions
 ----------------------------------
```

**Comment by vishwakftw** [comment]
There also seems to be a few formatting errors in the `one_hot` doc string. Could you fix that too, along with the function change in the example?

- The `Examples::` needs to be changed to `Examples:`
- Removal of new lines in the last example.

**Comment by krishnakalyan3** [comment]
@vishwakftw thank you so much for the feedback.

**Comment by vishwakftw** [comment]
Please revert the change wrt the function in add_docstr in torch.nn.functional.

**Comment by krishnakalyan3** [comment]
@vishwakftw I have reverted back the doc string change. However the docs string do not seem to render.

**Comment by vishwakftw** [comment]
Why do you think the docs aren't rendering? It renders for me.

Also, it'd be great if you could perform one more change:

```diff
diff --git a/torch/nn/functional.py b/torch/nn/functional.py
index 95945916e..9beb22cb6 100644
--- a/torch/nn/functional.py
+++ b/torch/nn/functional.py
@@ -2854,11 +2854,11 @@ Arguments:
         value in the input tensor.
 
 Returns:
-    Tensor: LongTensor that has one more dimension with 1 values at the
-        index of last dimension indicated by the input, and 0 everywhere
-        else.
+    LongTensor that has one more dimension with 1 values at the
+    index of last dimension indicated by the input, and 0 everywhere
+    else.
```

image

**Review by vishwakftw — APPROVED** [review]
Thanks for fixing this. This looks good to me.

**Comment by krishnakalyan3** [comment]
@vishwakftw This is what is rendered

image

Commands I am using
```
cd docs
make clean
make html-stable
```

**Comment by krishnakalyan3** [comment]
Thank you for the detailed feedback and guidance :).

**Comment by vishwakftw** [comment]
@krishnakalyan3 did you reinstall after making the changes? If you haven't then this is expected.

**Review by zou3519 — APPROVED** [review]

**Comment by vishwakftw** [comment]
This can be closed now, it has been merged into master.

**Pull request closed without merging**